### PR TITLE
Add data attributes to Menu Item

### PR DESCRIPTION
**Pull request by astridx**
### Summary of Changes

Add attributes to Menu Item

### CODE REVIEW

File: libraries/src/Form/Rule/DataElementRule.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * Joomla! Content Management System
+ *
+ * @copyright  (C) 2020 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline libraries/src/Form/Rule/DataElementRule.php]:
```suggestion
 * @copyright  (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
```

**Comment by HLeithner** [comment]
I would not go this way, why don't use use custom fields?

**Comment by astridx** [comment]
> I would not go this way, why don't use use custom fields?

Thank you for your answer, but I don't understand what you mean by that. 
Custom fields are connected to content, right? How can I link them to menu items?
My solution at the moment is to override the output of mod_menu.

**Comment by chmst** [comment]
As @brianteeman wrote. Will be happy to test your PR if you describe the use-case and add a test instruction.

**Comment by bembelimen** [comment]
I moved it to the 5.1-dev branch, as it is a new feature.

**Comment by chmst** [comment]
We close this PR for now. If mor information and testing isntruction are adde, we can reopen.

**Pull request closed without merging**